src/app.ts

```diff
@@ -0,0 +1,48 @@
+import Vue from 'vue';
+import { sync } from 'vuex-router-sync';
+import BootstrapVue from 'bootstrap-vue';
+import axios from 'axios';
+import { createProxy } from 'vuex-class-component';
+import App from './App.vue';
+import createRouter from './router';
+import createStore from './store/store';
+import InitGlobalComponents from '@/components/global';
+import UserStore from './store/user.vuex';
+import MobileStore from './store/mobile.vuex';
+import PageData from './store/page-data.vuex';
+
+Vue.use(BootstrapVue);
+
+InitGlobalComponents();
+
+Vue.config.productionTip = false;
+
+export default function createApp() {
+  const router = createRouter();
+  const store = createStore();
+
+  // Types for properties added to Vue (prototype) are in types/vue.d.ts
+  // Make sure to update that if you change what is included here!
+
+  const axiosInstance = axios.create({
+    withCredentials: true,
+  });
+  Vue.$http = axiosInstance;
+  Vue.prototype.$http = axiosInstance;
+
+  Vue.prototype.$vxm = {
+    user: createProxy(store, UserStore),
+    mobile: createProxy(store, MobileStore),
+    pageData: createProxy(store, PageData),
+  };
```

Comment:
I think this should be a non-default export of `store.ts`, so that we don't have to re-declare the modules on two different files.

Reply:
Issue is, we can't - the store instance is created within `createStore` (since we have to create fresh instances for SSR rather than creating it once when the file is first imported), so there's no way to export it. I also don't think it would be smart to export a default null variable that gets filled each time createStore is called, as I think there could be multiple instances at once when there are multiple SSR requests at the same time.

Reply:
I see...
Then maybe we can return it in the return value of createStore (by making it return an object of the store and the proxies)?
That would be weird to use though...

Reply:
It would still need to be added to Vue.prototype in able to be accessible by all components...

Reply:
When we would call createStore, we would save it into `store` and `proxies` variables, and then pass each of them to the right place.

Reply:
I mean all that gains us is defining it in `store.ts`... which I guess is good?